### ISSUE TITLE
parser: .d.ts should allow a function argument named `arguments`?

### ISSUE THREAD

**Issue by connorshea** [oxc-project/oxc#17776]
This is goofy, but I guess typescript technically allows it in `.d.ts` files:

```ts
declare global {
	interface TimeoutHandle { readonly _: never; /* this is a trick that seems needed to prevent direct number assignment */ }
	type Timeout = TimeoutHandle;
	function setTimeout(handler: string | Function, timeout?: number, ...arguments: any[]): Timeout;
}
```

As can be seen in the VS Code repo here: https://github.com/microsoft/vscode/blob/9ea3725fc61e6dd0cb3a1beedeb36c110e71b9bd/src/typings/base-common.d.ts#L25

[TS Playground example](https://www.typescriptlang.org/play/?ts=5.9.3&filetype=d.ts#code/PQKgtBldO38MU5EBQACE70GED2ADgJ4BOAlgOYAWALugBQDGAlOgLJmMl4DOeAZnXwkCeEgEMaZPADsAdOgCCAG2Xpy1Gj3UBTHjpIA3HQBM5GLOgAynHTP0n0AVxkmD6GlR3sAkgBVrW3sdBQBlHW8bRjt9ORoADzoyGQ8vdAJuACsdRjpuPDp+MXRlIP10ZKKSAFtJaXkLFCbmltakEGBUVGBgdAARHOVxEm8aIgI9VMl0AHdvDLwAI28qjzwK6oJlHWq7OicpNXFXYGLRZSJ+MlV0fhdc+p4ut0YhkfQKZSXxNQBvLoAkD10JAPGRdngDuheskaAZDD8GOIEddxItthUUj81IxZHDEjwADToRZQkxkfj8Ax7dA8SgySROEbaZLoGR4NxyTLaQzaOaLZiA2EGfjiaLoPzgnSQmgACWOJgxv104hMsgu6AA+gAuNk6YwkADc0KwnjILO04g85EYAGspnR9DttDIIm5HDR1hl9TTySNcmynNVliR0OIeHSKDJdjI6B10ABfVAAsYTCVSmXoAC86YhB3lrm2huTdxkD1ktJ0NEleZo9CoCu2JF1PBo5BkFHQAB90AAxe5SWTEqS1gD8upkQZDxLks+GFCDex4uuORAA2gBdZi6mvSg7FpOoHTxUQkOjKpNAA)

Discovered because of oxc-ecosystem-ci having 1 error in the vscode repo:

```
  × Cannot assign to 'arguments' in strict mode
    ╭─[src/typings/base-common.d.ts:25:71]
 24 │     type Timeout = TimeoutHandle;
 25 │     function setTimeout(handler: string | Function, timeout?: number, ...arguments: any[]): Timeout;
    ·                                                                          ─────────
 26 │     function clearTimeout(timeout: Timeout | undefined): void;
    ╰────

Found 561975 warnings and 1 error.
Finished in 1.2s on 6304 files with 352 rules using 8 threads.
```
